### PR TITLE
Add AUTHORS file

### PR DESCRIPTION
**Pull request by arkid15r**
## Proposed change

This PR adds the AUTHORS file generated by a script based on PH commits metadata.

## Type of change

- [ ] New country/market holidays support (thank you!)
- [ ] Supported country/market holidays update (calendar discrepancy fix, localization)
- [x] Existing code/documentation/test/process quality improvement (best practice, cleanup, refactoring, optimization)
- [ ] Dependency update (version deprecation/upgrade)
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] Breaking change (a code change causing existing functionality to break)
- [ ] New feature (new python-holidays functionality in general)

## Checklist

- [x] I've followed the [contributing guidelines][contributing-guidelines]
- [x] I've run `make pre-commit`, it didn't generate any changes
- [x] I've run `make test`, all tests passed locally

[contributing-guidelines]: https://github.com/vacanza/python-holidays/blob/dev/CONTRIBUTING.rst
[docs]: https://github.com/vacanza/python-holidays/tree/dev/docs/source

### CODE REVIEW

File: scripts/update_authors.py [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+    """
+    Generates authors based on the commits metadata.
+
+    Usage example: scripts/update_authors.py
+    """
+
+    @staticmethod
+    def run():
+        """Update authors list."""
+        authors = set()
```
Comment (KJhellico) [inline scripts/update_authors.py:38]:
```suggestion
        with open(PROJECT_DIR / OUTPUT_FILENAME, "r", encoding="UTF-8") as input_file:
            lines = input_file.readlines()
        authors = set(name.strip() for name in lines[3:])
```
If we plan to run this script from CI, with partially fetched repo.

File: scripts/update_authors.py [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+                if " " not in name:
+                    continue
+
+                tokens = name.split()
+                # Capitalize first name and last name.
+                for idx in (0, -1):
+                    tokens[idx] = tokens[idx].capitalize()
+                authors.add(" ".join(tokens))
+
+        with open(PROJECT_DIR / OUTPUT_FILENAME, "w") as output_file:
```
Comment (KJhellico) [inline scripts/update_authors.py:53]:
```suggestion
        with open(PROJECT_DIR / OUTPUT_FILENAME, "w", encoding="UTF-8") as output_file:
```
Windows :)

File: AUTHORS [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+Edward Betts
+Eldar Mustafayev
+Emmanuel Arias
+Eugenio Panadero Maciá
+Fabian Affolter
+Felix Lee
+Filip Bednárik
+Gabriel L Martinez
+Gabriel Trabanco
+Gabriel Trabanco Llano
```
Comment (KJhellico) [inline AUTHORS:46]: It seems to be the same person.

File: AUTHORS [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+Pieter van der Westhuizen
+Piotr Staniów
+Raphael Borg Ellul Vincenti
+Raychel Mattheeuw
+Reinaldo Ramos
+Robert Frazier
+Robert Schmidtke
+Robert Tran
+Robin Emeršič
+Ryan Mccrory
```
Comment (KJhellico) [inline AUTHORS:113]: McCrory

File: AUTHORS [hunk 5]
```diff
@@ -0,0 +1,133 @@
...
+Arkadii Yakovets
+Artem Tserekh
+Bailey Thompson
+Ben Collerson
+Ben Letham
+Benjamin Lucas Wacha
+Bernhard M. Wiedemann
+Carlos Rocha
+Chanran Kim
+Chris Mckeague
```
Comment (KJhellico) [inline AUTHORS:24]: Should be "McKeague"

File: AUTHORS [hunk 6]
```diff
@@ -0,0 +1,133 @@
...
+Gabriel L Martinez
+Gabriel Trabanco
+Gabriel Trabanco Llano
+Giedrius Mauza
+Gordon Inggs
+Greg Rafferty
+Győző Papp
+Heikki Orsila
+Henrik Sozzi
+Hugh Mcnamara
```
Comment (KJhellico) [inline AUTHORS:53]: McNamara

Comment (arkid15r) [reply AUTHORS]: ✔️

File: AUTHORS (hunk 4 above)
Comment (arkid15r) [reply AUTHORS]: ✔️

**Review by arkid15r — COMMENTED** [review]
Thanks for reviewing this! I kept the AUTHORS file and removed the update script as we decided to leave it to contributors to update AUTHORS with their names (and emails or other identifiers if they want).

**Review by KJhellico — APPROVED** [review]
LGTM.

**Pull request merged**